moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useSearch.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useMemo } from "react";
+
+/**
+ * Attempts to retrieve a valid graphql search value, for example when searching on an
+ * integer/float field but providing it a string, this function returns the value configured
+ * in the invalidValueDefault field in the search object, or null.
+ * @param {string} column - The name of the column to search
+ * @param {*} value - The value in question
+ * @returns {*} - The value output
+ */
+export const getSearchValue = (column, value, queryConfig) => {
+  // Retrieve the type of field (string, float, int, etc)
+  const type = queryConfig.columns[column].type.toLowerCase();
+  // Get the invalidValueDefault in the search config object
+  const invalidValueDefault =
+    queryConfig.columns[column].search?.invalidValueDefault ?? null;
+  // If the type is number of float, attempt to parse as such
+  if (["number", "float", "double"].includes(type)) {
+    value = Number.parseFloat(value) || invalidValueDefault;
+  }
+  // If integer, attempt to parse as integer
+  if (["int", "integer"].includes(type)) {
+    value = Number.parseInt(value) || invalidValueDefault;
+  }
+  // Any other value types are pass-through for now
+  return value;
+};
+
+export const useSearch = ({ queryConfig, defaultSearchTerm }) => {
+  const [searchTerm, setSearchTerm] = useState(defaultSearchTerm ?? "");
+
+  const searchWhereString = useMemo(() => {
```

Comment:
Same here - copypasta. The only change was to the return value that fits into the query in `useProjectListViewQuery`.